diff --git a/rhel/openvswitch.spec.in b/rhel/openvswitch.spec.in
index b0383ed..d5fbfe9 100644
--- a/rhel/openvswitch.spec.in
+++ b/rhel/openvswitch.spec.in
@@ -21,7 +21,7 @@
 # testing out of tree kernel modules the appropriate openvswitch-kmod
 # package should be installed first.
 
-Name: openvswitch
+Name: openvswitch-all
 Summary: Open vSwitch daemon/database/utilities
 Group: System Environment/Daemons
 URL: http://www.openvswitch.org/
@@ -30,7 +30,7 @@ Version: @VERSION@
 
 License: ASL 2.0
 Release: 1
-Source: openvswitch-%{version}.tar.gz
+Source: openvswitch-all-%{version}.tar.gz
 Buildroot: /tmp/openvswitch-rpm
 Requires: logrotate, hostname, python >= 2.7, python-six
 BuildRequires: python-six
@@ -39,7 +39,7 @@ BuildRequires: checkpolicy, selinux-policy-devel
 BuildRequires: autoconf, automake, libtool
 BuildRequires: python3-sphinx
 BuildRequires: unbound-devel
-BuildRequires: unwind-devel
+BuildRequires: libunwind-devel
 
 %bcond_without check
 %bcond_with check_datapath_kernel
@@ -49,12 +49,12 @@ Open vSwitch provides standard network bridging functions and
 support for the OpenFlow protocol for remote per-flow control of
 traffic.
 
-%package devel
-Summary:        Open vSwitch development package
-Group:          Development/Libraries
+#%package devel
+#Summary:        Open vSwitch development package
+#Group:          Development/Libraries
 
-%description devel
-This package provides openvswitch headers and libopenvswitch for developers.
+#%description devel
+#This package provides openvswitch headers and libopenvswitch for developers.
 
 %package selinux-policy
 Summary: Open vSwitch SELinux policy
@@ -266,7 +266,7 @@ exit 0
 /var/lib/openvswitch
 /var/log/openvswitch
 
-%files devel
+#%files devel
 %{_libdir}/lib*.so
 %{_libdir}/lib*.a
 %{_libdir}/pkgconfig
