From 0f54649d688da2ee541b6fd08e285f416bacce83 Mon Sep 17 00:00:00 2001
From: Pravin B Shelar <pbshelar@fb.com>
Date: Sun, 28 Jun 2020 08:02:10 +0000
Subject: [PATCH 12/19] ovs: datapath enable kernel 5.6

---
 acinclude.m4 | 2 +-
 manpages.mk  | 3 ---
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index 5842ce758..c765fc315 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -151,7 +151,7 @@ AC_DEFUN([OVS_CHECK_LINUX], [
     AC_MSG_RESULT([$kversion])
 
     if test "$version" -ge 5; then
-       if test "$version" = 5 && test "$patchlevel" -le 0; then
+       if test "$version" = 5 && test "$patchlevel" -le 6; then
           : # Linux 5.x
        else
           AC_ERROR([Linux kernel in $KBUILD is version $kversion, but version newer than 5.0.x is not supported (please refer to the FAQ for advice)])
diff --git a/manpages.mk b/manpages.mk
index dc201484c..54a3a82ad 100644
--- a/manpages.mk
+++ b/manpages.mk
@@ -104,7 +104,6 @@ utilities/bugtool/ovs-bugtool.8: \
 utilities/bugtool/ovs-bugtool.8.in:
 lib/ovs.tmac:
 
-
 utilities/ovs-dpctl-top.8: \
 	utilities/ovs-dpctl-top.8.in \
 	lib/ovs.tmac
@@ -155,8 +154,6 @@ lib/common-syn.man:
 lib/common.man:
 lib/ovs.tmac:
 
-lib/ovs.tmac:
-
 utilities/ovs-testcontroller.8: \
 	utilities/ovs-testcontroller.8.in \
 	lib/common.man \
-- 
2.17.1

