From aad8697575d90b1763c31ad06986be0550917aac Mon Sep 17 00:00:00 2001
From: Jacky Tian <xjtian@fb.com>
Date: Mon, 30 Mar 2020 22:28:45 -0700
Subject: [PATCH] Fix get_x509_serial for long serial numbers

---
 src/shrpx_tls.cc | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/src/shrpx_tls.cc b/src/shrpx_tls.cc
index 746311f0..83b148a6 100644
--- a/src/shrpx_tls.cc
+++ b/src/shrpx_tls.cc
@@ -2030,17 +2030,6 @@ StringRef get_x509_issuer_name(BlockAllocator &balloc, X509 *x) {
 #endif /* !WORDS_BIGENDIAN */
 
 StringRef get_x509_serial(BlockAllocator &balloc, X509 *x) {
-#if OPENSSL_1_1_API && !defined(OPENSSL_IS_BORINGSSL)
-  auto sn = X509_get0_serialNumber(x);
-  uint64_t r;
-  if (ASN1_INTEGER_get_uint64(&r, sn) != 1) {
-    return StringRef{};
-  }
-
-  r = bswap64(r);
-  return util::format_hex(
-      balloc, StringRef{reinterpret_cast<uint8_t *>(&r), sizeof(r)});
-#else  // !OPENSSL_1_1_API || OPENSSL_IS_BORINGSSL
   auto sn = X509_get_serialNumber(x);
   auto bn = BN_new();
   auto bn_d = defer(BN_free, bn);
@@ -2052,8 +2041,7 @@ StringRef get_x509_serial(BlockAllocator &balloc, X509 *x) {
   auto n = BN_bn2bin(bn, b.data());
   assert(n <= 20);
 
-  return util::format_hex(balloc, StringRef{std::begin(b), std::end(b)});
-#endif // !OPENSSL_1_1_API
+  return util::format_hex(balloc, StringRef{b.data(), static_cast<size_t>(n)});
 }
 
 namespace {
-- 
2.21.0

