From 3201e906d96e8b0dcb2e7f5912d96990b294ef2e Mon Sep 17 00:00:00 2001
From: Pravin B Shelar <pbshelar@fb.com>
Date: Tue, 21 Sep 2021 22:39:41 +0000
Subject: [PATCH 21/21] datapath: update version

---
 debian/changelog | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/changelog b/debian/changelog
index ed5b127e5..d48ede187 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,4 +1,4 @@
-openvswitch (2.15.2-1) unstable; urgency=low
+openvswitch (2.15.2-5) unstable; urgency=low
    [ Open vSwitch team ]
    * New upstream version
 
-- 
2.25.1

