diff --git a/third_party/build/bin/grpc_build.sh b/third_party/build/bin/grpc_build.sh
index 45e825c7c..ca919bddb 100755
--- a/third_party/build/bin/grpc_build.sh
+++ b/third_party/build/bin/grpc_build.sh
@@ -22,7 +22,7 @@ set -e
 SCRIPT_DIR="$(dirname "$(realpath "$0")")"
 source "${SCRIPT_DIR}"/../lib/util.sh

-GIT_VERSION=1.15.0
+GIT_VERSION=1.38.0
 ITERATION=3
 VERSION="${GIT_VERSION}"-"${ITERATION}"
 PKGNAME=grpc-dev
@@ -65,12 +65,13 @@ git submodule update --init
 sed -i 's/.usr.local$/\/tmp\/build-grpc-dev\/install\/usr\/local/' Makefile

 # build and install grpc
+mkdir -p cmake/build
+cd cmake/build
+cmake ../..
 make -j$(nproc)
 make install

-# HACK see https://github.com/grpc/grpc/issues/11868
-# package still links to libgrpc++.so.1 even though libgrpc++.so.6 is needed
-ln -sf ${WORK_DIR}/install/usr/local/lib/libgrpc++.so."${GIT_VERSION}" ${WORK_DIR}/install/usr/local/lib/libgrpc++.so.1
+mkdir /tmp/build-grpc-dev/install

 # packaging
 PKGFILE="$(pkgfilename)"
